openeogeotrellis/integrations/identity.py

```diff
@@ -77,6 +77,13 @@ def instance(cls) -> IDPTokenIssuer:
         else:
             return cls()
 
+    def _hard_reset(self) -> None:
+        """
+        This is only for testing purpose because different tests can actually require different idp config.
```

Comment:
can't this be solved through config override helpers as used in other tests?

Reply:
This component allows for live-reloading as long as config is valid.

If invalid config is put in place fatals are logged but no actual failure is being done as that could give an availability issue. For testing however it is important to be able to reset to cases where there is no config as well but such a config change won´t make changes because of the behavior detailed above. So I don´t see how to do this with config